mage_ai/cluster_manager/workspace/cloud_run.py

```diff
@@ -0,0 +1,69 @@
+import os
+
+import yaml
+
+from mage_ai.cluster_manager.config import CloudRunWorkspaceConfig
+from mage_ai.cluster_manager.constants import GCP_PROJECT_ID
+from mage_ai.cluster_manager.gcp.cloud_run_service_manager import CloudRunServiceManager
+from mage_ai.cluster_manager.workspace.base import Workspace
+from mage_ai.shared.hash import merge_dict
+
+
+class CloudRunWorkspace(Workspace):
+    def __init__(self, name: str):
+        super().__init__(name)
+        self.cloud_run_service_manager = CloudRunServiceManager(
+            self.config.project_id or os.getenv(GCP_PROJECT_ID),
+            self.config.path_to_credentials or os.getenv('path_to_keyfile'),
+            region=self.config.region or os.getenv('GCP_REGION'),
+        )
+
+    @property
+    def config(self) -> CloudRunWorkspaceConfig:
+        if os.path.exists(self.config_path):
+            return CloudRunWorkspaceConfig.load(config_path=self.config_path)
```

Comment:
nit: to avoid duplicate file read, I would suggest you caching the config (e.g. `self._config = config`) after the first read.
same in other classes.